### ISSUE TITLE
Memory leak under Python 3 when writing to shared memory

### ISSUE THREAD

**Issue by osvenskan** [osvenskan/sysv_ipc#7]
User Andy reported a memory leak when writing to `sysv_ipc.SharedMemory` in a loop. (Sample code attached.) They're right. This code works fine under Python 2, but under Python 3 `SharedMemory_write()` (in the `sysv_ipc` C code) is passed a `PyBuffer`, and the Python doc says, 
> you have to call PyBuffer_Release() after you have finished processing the data (or in any early abort case).

I'm not calling `PyBuffer_Release()`, so it's getting leaked on every call. Interestingly, passing a hardcoded byte string instead of a `numpy` object doesn't demonstrate the problem, I assume because Python is caching/re-using the object somehow whereas `numpy` mallocs a new array every time.

```
import numpy as np
import sysv_ipc
import resource

if __name__ =='__main__':
    queue = sysv_ipc.SharedMemory(None, flags=sysv_ipc.IPC_CREX, size=128, mode=0o600)
    try:
        while True:
            queue_info = np.array([5000, 64, 0, 0, 0])
            #queue_info = b'askdfhkjasdfkasdhfkhfkhflkjhasfhds'
            queue.write(queue_info)
            print('Memory usage : %s (kb)' % resource.getrusage(resource.RUSAGE_SELF).ru_maxrss)

    finally:
        queue.remove()
```

**Comment by osvenskan** [comment]
The same pattern exists in `MessageQueue_send()`, so it probably exhibits the same bug.

**Comment by osvenskan** [comment]
`MessageQueue_send()` doesn't exhibit the same bug because I coded it properly --
https://github.com/osvenskan/sysv_ipc/blob/develop/mq.c#L564

Glad I got that one right. Too bad that I got the shared mem implementation wrong. `:-/`

**Comment by osvenskan** [comment]
Fixed by PR #9 and included in release 1.0.1

**Comment by isghe** [comment]
I found, i think a similar memory leak, while reading (I am using `python3.8`):
``` python
                # Attach to the shared memory segment
                shm:SharedMemory = sysv_ipc.SharedMemory(shm_key)
                # Read the memory content
                shm_data = shm.read()
```
i will investigate in the next days,
thanks
